src/Microsoft.VisualStudio.Editors/Common/Utils.vb

```diff
@@ -314,36 +314,64 @@ Namespace Microsoft.VisualStudio.Editors.Common
+                OrElse TypeOf ex Is AccessViolationException
+        End Function
 
         ''' <summary>
-        ''' Given an exception, returns True if it is an "unrecoverable" exception.
+        ''' Logs the given exception and returns whether or not this is a "recoverable" exception, i.e. exception can be ignored.
         ''' </summary>
-        ''' <param name="ex">The exception to check rethrow if it's unrecoverable</param>
-        ''' <param name="IgnoreOutOfMemory">If True, out of memory will not be considered unrecoverable.</param>
+        ''' <param name="ex">The exception to log and check if it is recoverable.</param>
+        ''' <param name="exceptionEventDescription">Additional description for the cause of the exception.</param>
+        ''' <param name="throwingComponentName">Name of the component that threw that exception, generally the containing type name.</param>
+        ''' <param name="debugFail">If True, then invoke Debug.Fail for recoverable exception.</param>
+        ''' <param name="debugWriteLine">If True, then invoke Debug.WriteLine for recoverable exception.</param>
+        ''' <param name="considerExceptionAsRecoverable">If True, then the given exception is always considered recoverable.
+        ''' Otherwise, invokes <see cref="IsUnrecoverableException(Exception, Boolean)"/> to determine if exception is recoverable or not.</param>
+        ''' <param name="ignoreOutOfMemory">If True, out of memory will be considered recoverable.</param>
         ''' <remarks></remarks>
-        Public Function IsUnrecoverable(ByVal ex As Exception, Optional ByVal IgnoreOutOfMemory As Boolean = False) As Boolean
-            If TypeOf ex Is NullReferenceException _
-                OrElse (Not IgnoreOutOfMemory AndAlso TypeOf ex Is OutOfMemoryException) _
-                OrElse TypeOf ex Is StackOverflowException _
-                OrElse TypeOf ex Is System.Threading.ThreadAbortException _
-                OrElse TypeOf ex Is AccessViolationException _
-            Then
-                Return True
+        Public Function ReportWithoutCrash(ByVal ex As Exception,
+                                           ByVal exceptionEventDescription As String,
+                                           Optional ByVal throwingComponentName As String = "general",
+                                           Optional ByVal debugFail As Boolean = False,
+                                           Optional ByVal debugWriteLine As Boolean = False,
+                                           Optional ByVal considerExceptionAsRecoverable As Boolean = False,
+                                           Optional ByVal ignoreOutOfMemory As Boolean = False) As Boolean
+            Debug.Assert(ex IsNot Nothing)
+            Debug.Assert(Not String.IsNullOrEmpty(throwingComponentName))
+            Debug.Assert(Not String.IsNullOrEmpty(exceptionEventDescription))
+
+            TelemetryService.DefaultSession.PostFault(
```

Comment:
More information about this VSTelemetry API over [here](http://vstelwiki/wiki/Fault_Event)